cap-primitives/src/winx/fs/metadata_ext.rs

```diff
@@ -0,0 +1,94 @@
+#![allow(clippy::useless_conversion)]
+
+use std::fs;
+
+#[derive(Debug, Clone)]
+pub(crate) struct MetadataExt {
+    pub(crate) file_attributes: u32,
+    pub(crate) creation_time: u64,
+    pub(crate) last_access_time: u64,
+    pub(crate) last_write_time: u64,
+    pub(crate) file_size: u64,
+    pub(crate) volume_serial_number: Option<u32>,
+    pub(crate) number_of_links: Option<u32>,
+    pub(crate) file_index: Option<u64>,
+}
+
+impl MetadataExt {
+    /// Constructs a new instance of `Self` from the given `std::fs::Metadata`.
+    #[inline]
+    pub(crate) fn from_std(std: fs::Metadata) -> Self {
+        use std::os::windows::fs::MetadataExt;
+        Self {
+            file_attributes: std.file_attributes(),
+            creation_time: std.creation_time(),
+            last_access_time: std.last_access_time(),
+            last_write_time: std.last_write_time(),
+            file_size: std.file_size(),
+            volume_serial_number: std.volume_serial_number(),
+            number_of_links: std.number_of_links(),
+            file_index: std.file_index(),
+        }
+    }
+
+    /// Determine if `self` and `other` refer to the same inode on the same device.
+    #[cfg(debug_assertions)]
+    pub(crate) fn is_same_file(&self, other: &Self) -> bool {
+        // From [MSDN]:
+        // The identifier (low and high parts) and the volume serial number uniquely identify a file on a single computer.
+        // To determine whether two open handles represent the same file, combine the identifier and the volume serial
+        // number for each file and compare them.
```

Comment:
Could you reflow this paragraph to fit in 100 columns?

Reply:
Sure thing!